python/amici/__init__.py

```diff
@@ -84,41 +106,43 @@ def _get_commit_hash():
 hdf5_enabled = False
 has_clibs = False
 
-try:
-    # Try importing AMICI SWIG-interface with HDF5 enabled
-    from . import amici
-    from .amici import *
-    hdf5_enabled = True
-    has_clibs = True
-except ModuleNotFoundError:
-    # import from setuptools or installation with `--no-clibs`
-    pass
-except (ImportError, AttributeError) as e:
-    # No such module exists or there are some dynamic linking problems
-    if isinstance(e, AttributeError) or "cannot import name" in str(e):
-        # No such module exists (ImportError),
-        #  or python tries to import a HDF5 function from the non-hdf5
-        #  swig interface (AttributeError):
-        #  try importing AMICI SWIG-interface without HDF5
-        try:
-            from . import amici_without_hdf5 as amici
-            from .amici_without_hdf5 import *
-            has_clibs = True
-        except ModuleNotFoundError:
-            # import from setuptools or installation with `--no-clibs`
-            pass
-        except ImportError as e:
-            if "cannot import name" in str(e):
-                # No such module exists
-                # this probably means, the model was imported during setuptools
-                # `setup` or after an installation with `--no-clibs`.
+if not _imported_from_setup():
+    try:
+        # Try importing AMICI SWIG-interface with HDF5 enabled
+        from . import amici
+        from .amici import *
+        hdf5_enabled = True
+        has_clibs = True
+    except ModuleNotFoundError:
+        # import from setuptools or installation with `--no-clibs`
```

Comment:
can we also detect this using frame inspection, this would reduce remaining complexity substantially.

Reply:
The first part of the comment is now obsolete (will remove). The second part is maybe misleading, it's not during installation, it's after being installed with `--no-clibs`. We could circumvent that by writing a "configuration" file during installation to include whether we have HDF5 support or have the extension at all. That would clean up that mess quite well :)

Reply:
How about `rm amici_without_hdf5.py` if hdf5 was detected during installation and `rm amici.py;mv amici_without_hdf5.py amici.py` if it was not?

Reply:
Sounds easier. Should do the job as well.
That this removes the possibility of rebuilding the extension from the installed package later on is probably not of practical relevance...